tests/DomCrawler/CrawlerTest.php

```diff
@@ -156,6 +156,50 @@ public function testSiblings(callable $clientFactory): void
         $this->assertSame(['Main', 'Sibling 2', 'Sibling 3'], $texts);
     }
 
+    /**
+     * @dataProvider clientFactoryProvider
+     */
+    public function testMatches(callable $clientFactory): void
+    {
+        $crawler = $this->request($clientFactory, '/basic.html');
+        $p = $crawler->filter('#a-sibling');
+
+        $this->assertTrue($p->matches('#a-sibling'));
+        $this->assertTrue($p->matches('p'));
+        $this->assertTrue($p->matches('.foo'));
+        $this->assertFalse($p->matches('#other-id'));
+        $this->assertFalse($p->matches('div'));
+        $this->assertFalse($p->matches('.bar'));
+    }
+
+    /**
+     * @dataProvider clientFactoryProvider
+     */
+    public function testClosest(callable $clientFactory): void
```

Comment:
Test and fixture copied from https://github.com/symfony/dom-crawler/blob/5.4/Tests/AbstractCrawlerTest.php#L937